File: wp-2019/src/main/java/ukim/mk/finki/konstantin/bogdanoski/wp/service/PizzaService.java
package ukim.mk.finki.konstantin.bogdanoski.wp.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ukim.mk.finki.konstantin.bogdanoski.wp.model.Pizza;

/**
 * @author Konstantin Bogdanoski (konstantin.b@live.com)
 */
public interface PizzaService extends BaseEntityCrudService<Pizza> {
    public Pizza findByName(String name);

    public Page<Pizza> findPaginated(Pageable pageable);
}

File: frontend/src/service/pizzaService.js
/**
 * @author Konstantin Bogdanoski (konstantin.b@live.com)
 */
import React, {Component} from "react";
import axios from '../custom-axios/axios.js';
import qs from 'qs';

const PizzaService = {
    fetchPizzas: () => {
        return axios.get("/pizzas");
    },

    fetchPizzasPaged: (page, pageSize) => {
        return axios.get("/pizzas", {
            headers: {
                'page': page, 'page-size': pageSize
            }
        })
    },

    addPizza: (pizza) => {
        const data = {
            ...pizza,
        };
        const formParams = qs.stringify(data);
        return axios.post("/pizzas", formParams, {
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded',
            }
        });
    },

    deletePizza: (pizzaID) => {
        return axios.delete("/pizzas/" + pizzaID);
    },

    findPizzas: (searchTerm) => {
        return axios.get(`/pizzas?searchTerm=${searchTerm}`);
    }
};

export default PizzaService;